src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -939,9 +939,24 @@ class CaseExpressionSegment(BaseSegment):
                 Sequence(
                     Ref.keyword('NOT', optional=True),
                     'BETWEEN',
+                    # In a between expression, we're restricted to arithmetic operations
+                    # which means we can look for some binary operations without worrying
+                    # about the binary ones (and therefore AND).
```

Comment:
> look for some binary operations without worrying about the binary ones

The use of "binary" here twice confuses me 🤔 could you clarify this (💯 love your detailed commenting style btw)

Reply:
I don't think what I've written actually makes sense! Sorry about that - clearly written while sleepy 😴 .

I shall reword it to be more useful.